frappe/core/doctype/user/user.py

```diff
@@ -855,8 +855,12 @@ def check_roles_added(self):
 			indicator="orange",
 			primary_action={
 				"label": _("Add Roles"),
-				"client_action": "frappe.set_route",
-				"args": ["Form", self.doctype, self.name],
+				"client_action": "eval",
```

Comment:
Works fine but wondering if there's a better way to handle this.

Reply:
Somewhat like https://github.com/frappe/erpnext/blob/develop/erpnext/stock/utils.py#L582-L585
https://github.com/frappe/erpnext/blob/develop/erpnext/public/js/utils.js#L77-L79